code/game/machinery/doors/firedoor.dm

```diff
@@ -130,15 +131,38 @@
 		log_game("[key_name(user)] [welded ? "welded":"unwelded"] firedoor [src] with [W] at [AREACOORD(src)]")
 		update_appearance()
 
-/obj/machinery/door/firedoor/try_to_crowbar(obj/item/I, mob/user)
+/// We check for adjacency when using the primary attack.
+/obj/machinery/door/firedoor/try_to_crowbar(obj/item/acting_object, mob/user)
 	if(welded || operating)
 		return
 
 	if(density)
+		being_held_open = TRUE
 		open()
+		RegisterSignal(user, COMSIG_MOVABLE_MOVED, .proc/handle_held_open_adjacency)
+		RegisterSignal(user, COMSIG_LIVING_SET_BODY_POSITION, .proc/handle_held_open_adjacency)
```

Comment:
Both these have to have SIGNAL_HANDLER on them.

https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#signal-handlers

Reply:
I added the SIGNAL HANDLER to it, but it runtimes since close() needs to sleep. I circumvented it by having it be on a 1ds callback timer though. I hope this is okay?

Reply:
Just use INVOKE_ASYNC.